src/server/scripts/Outland/zone_hellfire_peninsula.cpp

```diff
@@ -420,10 +420,481 @@ class npc_fel_guard_hound : public CreatureScript
+                    me->AddAura(SPELL_JULES_GOES_PRONE, me);
+                    me->AddAura(SPELL_JULES_THREATENS_AURA, me);
+
+                    me->SetCanFly(true);
+                    me->SetSpeed(MOVE_RUN, 0.2f);
+                    
+                    me->SetFacingTo(3.207566f);
+                    me->GetMotionMaster()->MoveJump(exorcismPos[2], 2.0f, 2.0f);
+
+                    events.ScheduleEvent(EVENT_SUMMON_SKULL, 10000);
+                    break;
+                case ACTION_JULES_FLIGH:
+                    circleRounds++;
+
+                    me->RemoveAura(SPELL_JULES_GOES_PRONE);
+
+                    me->AddAura(SPELL_JULES_GOES_UPRIGHT, me);
+                    me->AddAura(SPELL_JULES_VOMITS_AURA, me);
+
+                    wpreached = true;
+                    me->GetMotionMaster()->MovePoint(point, exorcismPos[point]);
+                    break;
+                case ACTION_JULES_MOVE_HOME:
+                    wpreached = false;
+                    me->SetSpeed(MOVE_RUN, 1.0f);
+                    me->GetMotionMaster()->MovePoint(11, exorcismPos[2]);
+
+                    events.CancelEvent(EVENT_SUMMON_SKULL);
+                    break;
+            }
+        }
+
+        void JustSummoned(Creature* summon) override
+        {
+            summons.Summon(summon);
+
+            if (summon->GetAI())
+                summon->GetMotionMaster()->MoveRandom(10.0f);
+        }
+
+        void MovementInform(uint32 type, uint32 id) override
+        {
+            if (type != POINT_MOTION_TYPE)
+                return;
+
+            if (id < 10)
+                wpreached = true;
+
+            if (id == 8)
+            {
+                for (uint8 i = 0; i < circleRounds; i++)
+                    DoSummon(NPC_FOUL_PURGE, exorcismPos[8]);
+            }
+
+            if (id == 10)
+            {
+                wpreached = true;
+                point = 3;
+                circleRounds++;
+            }
+        }
+
+        void UpdateAI(uint32 diff) override
+        {
+            if (wpreached)
+            {
+                me->GetMotionMaster()->MovePoint(point, exorcismPos[point]);
+                point++;
+                wpreached = false;
+            }
+
+            events.Update(diff);
+
+            while (uint32 eventId = events.ExecuteEvent())
+            {
+                switch (eventId)
+                {
+                    case EVENT_SUMMON_SKULL:
+                        uint8 summonCount = urand(1,3);
+
+                        for (uint8 i = 0; i < summonCount; i++)
+                            me->SummonCreature(NPC_DARKNESS_RELEASED, me->GetPositionX(), me->GetPositionY(), me->GetPositionZ() + 1.5f, 0, TEMPSUMMON_MANUAL_DESPAWN);
+
+                        events.ScheduleEvent(EVENT_SUMMON_SKULL, urand(10000, 15000));
+                        break;
+                }
+            }
+        }
+        
+        private:
+        EventMap events;
```

Comment:
Sorry, I forgot that you are referring the line directly above your comment. So I need to increase the indent (with 4 spaces) on "EventMap events;" if I understand you correctly this time?